pulsar-broker/src/main/java/org/apache/pulsar/broker/BookKeeperClientFactoryImpl.java

```diff
@@ -159,6 +159,7 @@ ClientConfiguration createBkClientConfiguration(MetadataStoreExtended store, Ser
                 bkConf.setProperty(bkExtraConfigKey, value);
             }
         });
+        bkConf.setWaitTimeoutOnBackpressureMillis(conf.getWaitTimeoutOnBackpressureMillis());
```

Comment:
Setting this configuration here would override an existing `bookkeeper_waitTimeoutOnBackpressureMillis` configuration in the `broker.conf` file. That could lead a user to unexpectedly get the Pulsar default value. I think we should move this line to line 152.